lib/vrl/cli/src/repl.rs

```diff
@@ -135,7 +163,26 @@ impl Hinter for Repl {
     type Hint = String;
 
     fn hint(&self, line: &str, pos: usize, ctx: &Context<'_>) -> Option<String> {
-        self.hinter.hint(line, pos, ctx)
+        if pos < line.len() {
+            return None;
+        }
+
+        // Check history first
+        if let Some(hist) = self.history_hinter.hint(line, pos, ctx) {
+            return Some(hist);
+        }
+
+        // Then check the other built-in hints
+        self.hints
+            .iter()
+            .filter_map(|hint| {
+                if pos > 0 && hint.starts_with(&line[..pos]) {
+                    Some(String::from(&hint[pos..]))
+                } else {
+                    None
+                }
+            })
+            .next()
```

Comment:
`filter_map(func).next()` could be replaced by `find_map(func)`, I think the meaning is more clear using `find_map` too.